packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -411,23 +411,19 @@ class _DefaultPub implements Pub {
 
         exitCode = result.exitCode;
       }
-    // The exception is rethrown, so don't catch only Exceptions.
-    } catch (exception) { // ignore: avoid_catches_without_on_clauses
-      if (exception is io.ProcessException) {
```

Comment:
This one is a kind of interesting story…

It used to be that this block called `status?.cancel();` before checking the exception's type, but then that was removed in #115801.

With that line out of the way, we can convert this to an `on` clause!